### ISSUE TITLE
Build fails with SSR turned off

### ISSUE THREAD

**Issue by francisfontoura** [nuxt/bridge#27]
###  Environment

- Operating System: `Linux`
- Node Version:     `v14.18.0`
- Nuxt Version:     `2.16.0-27226092.034b9901`
- Package Manager:  `Yarn`
- Bundler:          `Webpack`
- User Config:      `ssr`, `buildModules`
- Runtime Modules:  `-`
- Build Modules:    `@nuxt/bridge@3.0.0-27234503.d5127e9`

### Describe the bug and expected behavior

`nuxi build` fails with `ssr: false`. Output:
```
start Building server...                                                                                                                                               00:26:02

 ERROR  Rollup error: Could not load .nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '.nuxt/dist/server/server.mjs'


 ERROR  Nuxt Build Error: Could not load .nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '.nuxt/dist/server/server.mjs'
```

### Steps to reproduce

1. Set `ssr: false` in `nuxt.config.ts`.
2. Run (`npx`) `nuxi build` in shell.

### Additional context

My project doesn't need SSR, but needs "runtime config" with environment variables injected by Node.js.

**Comment by Yeganloo** [comment]
I hop fixing this bug be a priority for Nuxt team.
I know you are busy on the Nuxt3 it self but as a matter of fact Vuetify is one of most popular component frameworks for Vue and they are so slow on migrating to Vue3. Me my self am trying to replace it with pure css frameworks like tailwind but until then it is not possible for me to switch to Nuxt3. Meanwhile i have to make the hole project ready for Vue3 and Nuxt3 and it is lots of work! so i choose bridge as the best migration option. but after #320 i kinda stuck!

**Comment by HendrikJan** [comment]
I hope so too.
We are waiting for Vuetify 3 too before we can move to the real Nuxt 3 (really looking forward to that!).
Currently using `nuxi dev ` instead of `nuxi build` but that's not a solution when our production goes live 🙃.

**Comment by MartinX3** [comment]
Currently we use Nuxt2 and Vuetify with the composition API package, since even the bridge doesn't work without SSR.

Just the terminal tells me everytime, that I should migrate from the composition API package to nuxt bridge.


But anyway we should wait for Vue 2.7 which is a migration release for Vue 3.

**Comment by rodrigogs** [comment]
You guys should check this: https://www.youtube.com/watch?v=DE3HABBzUWY&ab_channel=RazorCXTechnologies

**Comment by MartinX3** [comment]
> You guys should check this: https://www.youtube.com/watch?v=DE3HABBzUWY&ab_channel=RazorCXTechnologies

How does that fix building without SSR?

And if you look at the roadmap vuetufy 3 beta is still missing many components

**Comment by crazymind** [comment]
Updating @nuxt/bridge solved this issue for me (Note: I had problems running my app locally without ssr and not building – but the error was the same).

This commit on the 13. April solved it:
https://github.com/nuxt/bridge/commit/77db7ed726d6e173f9d92fc29d0dcc033785b5aa

**Comment by jtc42** [comment]
> Updating @nuxt/bridge solved this issue for me (Note: I had problems running my app locally without ssr and not building – but the error was the same).
> 
> This commit on the 13. April solved it: [77db7ed](https://github.com/nuxt/bridge/commit/77db7ed726d6e173f9d92fc29d0dcc033785b5aa)

I upgraded to [3.0.0-27530716.c4f8671](https://www.npmjs.com/package/@nuxt/bridge-edge/v/3.0.0-27530716.c4f8671) which should include that commit but I still get the exact same error... Am I missing something?

**Comment by crazymind** [comment]
Can confirm that the ` nuxt generate` still fails. `nuxi dev` is working.

I made a fork with quick fix to hardcode disable SSR. This way I was able to build my app with Netlify. 
Really just a temporary fix, till the problem is solved properly. 

Delete: node_modules
Delete: yarn.lock
```
yarn remove @nuxt/bridge-edge
yarn add --dev crazymind/bridge
yarn
```

Sidenote: I received another build error **nitro preset was undefined** after fixing the server.mjs problem. I changed the nitroconfig preset to `nitro-dev` in this fork. Not quite sure if this causes some security issues on prod?
`preset: 'nitro-dev', // nuxt.options.dev ? 'nitro-dev' : undefined,`

Hope this also works for you guys🙏

**Comment by eLvErDe** [comment]
Hello,

I can confirm this is not working either for me with fresh nuxt-bridge install from today, I'm rolling back.
Moreover, disabling nitro in nuxt.config.ts does not really help as it seems to generate weird file in `.nuxt/dist/server` and `.nuxt/dist/client` but nothing that looks like being ready to be deployed on a webserver.

**Comment by rvmourik** [comment]
I created a work a round, that makes it possible to build without using a fork:

```js
  hooks: {
    'build:done': (builder) => {
      const extraFilePath = path.join(
        builder.nuxt.options.buildDir + '/dist/server',
        'server.mjs'
      );
      fs.writeFileSync(extraFilePath, 'export {};');
    }
  },
```

This will put a file in the proper directory just before rollup needs it.

**Comment by Yeganloo** [comment]
I can't believe an empty file could solve this problem!
Thanks a lot @rvmourik

**Comment by rvmourik** [comment]
> @rvmourik your solution worked for my nuxt-bridge SPA deployed on vercel.
> 
> I did then run into an error when attempting to access the app from any path that was not the root of the project - I had to install the `ufo` npm package the dependency was not in scope for any route of the SPA that wasn't the root for some reason but once the ufo npm package was added no problems.

Yes, that was another problem I also encountered. But that has nothing to do with the server files not being present _or_ skipped. I managed to resolve the `ufo` issue by adding it to a resolves in package.json and set it to 0.8.4. 

Related issue https://github.com/nuxt/bridge/issues/345#issuecomment-1158528662

**Comment by MartinX3** [comment]
At the end of `nuxt generate` it tells me
```
ℹ You can preview this build using node .output/server/index.mjs                                                                                                      nitro 11:06:55
Listening on http://localhost:3000/
```

Is that normal?
I just want a static build without SSR.

(`nuxi generate` doesn't work because of https://github.com/nuxt/bridge/issues/365)

**Comment by MartinX3** [comment]
It seems like the generator generates a broken .js file in the `_nuxt` folder?

`Uncaught SyntaxError: Invalid or unexpected token (at f1e478e.js:2:76798)`

I did autoformat the file in IntelliJ and find errors like this
```js
        reduceElements(t) {
            const e = this.Ï€ + this.p, [n, o] = r.b.sincos(t.inc), [c, m] = r.b.sincos(t.node - this.Ï€), d = new l.a
            return d.inc = Math.acos(o * this.cÎ·+n * this.sÎ·*
            m
        ),
            d.node = Math.atan2(n * c, this.cÎ·*
            n * m - this.sÎ·*
            o
        )
            +e, d.peri = Math.atan2(-this.sÎ·*
            c, n * this.cÎ·-o * this.sÎ·*
            m
        )
            +t.peri, d
        }
```

**Comment by danielroe** [comment]
@MartinX3 Would you raise a new issue with a reproduction? I think there's something unique about your setup to examine.

**Comment by MartinX3** [comment]
@danielroe I try to locate the problem.
As far as I see it might be the usage of a pinia store causing it.
